src/components/table/schedule/schedule-parts/base-row.component.tsx

```diff
@@ -55,35 +50,39 @@ export function ScheduleRowComponent({
       setSelectedCells([]);
     }
   }, [selectionMode]);
+
   const verboseDates = scheduleLogic?.sections.Metadata?.verboseDates;
 
-  function onContextMenu(cellIndex: number): void {
-    if (scheduleLogic) {
-      scheduleLogic.changeShiftFrozenState(index, cellIndex, setFrozenShifts);
-    }
-  }
+  const onContextMenu = useCallback(
+    (cellIndex: number): void => {
+      if (scheduleLogic) {
+        scheduleLogic.changeShiftFrozenState(index, cellIndex);
+      }
+    },
+    [scheduleLogic, index]
+  );
+
   function saveValue(newValue: string): void {
     if (sectionKey)
       scheduleLogic?.updateRow(
         sectionKey,
         index - 1,
         [...selectedCells, pointerPosition],
-        newValue,
-        (newDataRow) => {
-          setDataRowState(newDataRow);
-          onStateUpdate && onStateUpdate(newDataRow);
-        }
+        newValue
       );
     setSelectedCells([]);
   }
 
-  function isFrozen(cellInd: number): boolean {
-    return (
-      verboseDates?.[cellInd]?.isFrozen ||
-      !!frozenShifts.find((fS) => fS[0] === index && fS[1] === cellInd) ||
-      false
-    );
-  }
+  const isFrozen = useCallback(
```

Comment:
Przepisałem dość dużo funkcji na `useCallback` ponieważ inaczej one były tworzone od nowa przy każdym rederowaniu komponentu ( w tym sensie,że zmieniała się referencja ). Przez to komponenty, które dostawały te funkcje jako props'y, uważały, że to są nowe wartości i mogły wyrenrderować się od nowa. Tu jest  bardzo fajny filmik na ten temat:
https://www.youtube.com/watch?v=o-alRbk_zP0&t=622s

Tłumaczy dość dużo z tego co jest w MR :)